<?xml version="1.0" encoding="utf-8"?>

<launch>
    <arg name="sim" default="true" />
    <arg name="robot_name" default="ur5" />
    <arg name="group_name"   default="manipulator"/>
    <arg name="base_frame"   default="base_link"/>
    <!-- <arg name="cam_frame"    default="camera_color_optical_frame"/> -->

    <!-- UR5 config & MoveIt-->
    <group if="$(eval robot_name == 'ur5')">
        <include file="$(find ur_gazebo)/launch/ur5_bringup.launch" />
        <include file="$(find ur5_moveit_config)/launch/moveit_planning_execution.launch">
            <arg name="sim" default="$(arg sim)" />
        </include>
        <include file="$(find ur5_moveit_config)/launch/moveit_rviz.launch"/>
    </group>

    <!-- UR5e config & MoveIt-->
    <group if="$(eval robot_name == 'ur5e')">
        <include file="$(find ur_gazebo)/launch/ur5e_bringup.launch" />
        <include file="$(find ur5e_moveit_config)/launch/moveit_planning_execution.launch">
            <arg name="sim" default="$(arg sim)" />
        </include>
        <include file="$(find ur5e_moveit_config)/launch/moveit_rviz.launch"/>
    </group>

    <!-- <node name="check_methods" pkg="plan_robot_path" type="check_methods.py" output="screen" respawn="true"> -->
    <node name="rotate_one_joint" pkg="plan_robot_path" type="rotate_one_joint.py" output="screen" respawn="true">
        <param name="robot_name" value="$(arg robot_name)" />
        <param name="group_name" value="$(arg group_name)" />
        <param name="base_frame" value="$(arg base_frame)" />
    </node>
    
</launch>